core/src/main/java/com/datastax/oss/driver/api/core/session/SessionBuilder.java

```diff
@@ -130,8 +132,26 @@ public SelfT addTypeCodecs(TypeCodec<?>... typeCodecs) {
     return self;
   }
 
-  public SelfT addNodeStateListeners(NodeStateListener... newListeners) {
-    Collections.addAll(this.nodeStateListeners, newListeners);
+  /**
+   * Registers a node state listener to use with the session.
+   *
+   * <p>If the listener is specified programmatically with this method, it overrides the
+   * configuration (that is, the {@code metadata.node-state-listener.class} option will be ignored.
+   */
+  public SelfT withNodeStateListener(NodeStateListener nodeStateListener) {
```

Comment:
👍 makes sense to have builder methods for these based on our [discussion](https://github.com/datastax/java-driver/pull/989#discussion_r180159691) in #989.